dgraphtest/graphql.go

```diff
@@ -0,0 +1,227 @@
+	if value1.Type() != value2.Type() {
+		return false
+	}
+
+	// Compare the values of the interfaces
+	return reflect.DeepEqual(expected, actual)
+}
+
+func (hc *HTTPClient) SafelyUpdateGQLSchema(schema string, alphaPort string) *GqlSchema {
+	// first, make an initial probe to get the schema update counter
+	oldCounter := hc.RetryProbeGraphQL(alphaPort).SchemaUpdateCounter
+
+	fmt.Println("oldCounter>>>>>>>>>>>", oldCounter)
+
+	// update the GraphQL schema
+	gqlSchema := hc.AssertUpdateGQLSchemaSuccess(alphaPort, schema)
+
+	fmt.Println("gqlSchema>>>>>>>>>>>", gqlSchema)
+	// // now, return only after the GraphQL layer has seen the schema update.
+	// // This makes sure that one can make queries as per the new schema.
+	// AssertSchemaUpdateCounterIncrement(t, authority, oldCounter, headers)
+	// return gqlSchema
+	return nil
+}
+
+// **********************************************************************************************************
+
+func (hc *HTTPClient) probeGraphQL(alphaPort string) (*ProbeGraphQLResp, error) {
+
+	request, err := http.NewRequest("GET", "http://localhost:"+alphaPort+"/probe/graphql", nil)
+	if err != nil {
+		return nil, err
+	}
+	client := &http.Client{}
+	//request.Header = header
+	resp, err := client.Do(request)
+	if err != nil {
+		return nil, err
+	}
+
+	probeResp := ProbeGraphQLResp{}
+	if resp.StatusCode == http.StatusOK {
+		probeResp.Healthy = true
+	}
+
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	if err = json.Unmarshal(b, &probeResp); err != nil {
+		return nil, err
+	}
+	return &probeResp, nil
+}
+
+func (hc *HTTPClient) retryProbeGraphQL(alphaPort string) *ProbeGraphQLResp {
+	for i := 0; i < 10; i++ {
+		resp, err := hc.probeGraphQL(alphaPort)
+		if err == nil && resp.Healthy {
+			return resp
+		}
+		time.Sleep(time.Second)
+	}
+	return nil
+}
+
+func (hc *HTTPClient) RetryProbeGraphQL(alphaPort string) *ProbeGraphQLResp {
+	if resp := hc.retryProbeGraphQL(alphaPort); resp != nil {
+		return resp
+	}
+	debug.PrintStack()
+	//t.Fatal("Unable to get healthy response from /probe/graphql after 10 retries")
+	return nil
+}
+
+// **********************************************************************************************************
+
+// AssertUpdateGQLSchemaSuccess updates the GraphQL schema, asserts that the update succeeded and the
+// returned response is correct. It returns a *GqlSchema it received in the response.
+func (hc *HTTPClient) AssertUpdateGQLSchemaSuccess(alphaPort string, schema string) *GqlSchema {
+	// update the GraphQL schema
+	updateResp := hc.RetryUpdateGQLSchema(alphaPort, schema)
+	// sanity: we shouldn't get any errors from update
+	//RequireNoGQLErrors(updateResp)
+
+	// sanity: update response should reflect the new schema
+	var updateResult struct {
+		UpdateGQLSchema struct {
+			GqlSchema *GqlSchema
+		}
+	}
+	if err := json.Unmarshal(updateResp, &updateResult); err != nil {
+		debug.PrintStack()
```

Comment:
when we panic, we do not need to call debug.PrintStack()

Reply:
Resolved